core/src/main/kotlin/xtdb/IBufferPool.kt

```diff
@@ -1,12 +1,33 @@
 package xtdb
 
 import org.apache.arrow.memory.ArrowBuf
+import org.apache.arrow.vector.ipc.message.ArrowFooter
+import org.apache.arrow.vector.ipc.message.ArrowRecordBatch
 import xtdb.arrow.Relation
 import java.nio.ByteBuffer
 import java.nio.file.Path
 
 interface IBufferPool : AutoCloseable {
-    fun getBuffer(key: Path): ArrowBuf
+    /**
+     * Get the whole file as an on-heap ByteBuffer.
+     *
+     * This should only be used for small files (metadata) or testing purposes.
+     */
+    fun getByteBuffer(key: Path): ByteBuffer
```

Comment:
Apart from tests, it seems like the only usages you have of this would all prefer to have a byte-array, which is easier to get hold of from an ArrowBuf, and also more obviously on-heap